packages/frontend/src/context/StoreContext.tsx

```diff
@@ -202,13 +216,40 @@ export const StoreContextProvider = (
             setErc20Addr,
             setPublishedTagId,
             setFinalizedCarts,
+            setPubKeys,
+            walletAddress,
           );
         }
       }
     } catch (err) {
       console.error("error receiving events", err);
     }
   };
+  const verify = async (
+    _cartItems: Map<CartId, CartState>,
+    _pubKeys: `0x${string}`[],
+  ) => {
+    const addresses = _pubKeys.map((k) => {
+      return Address.fromPublicKey(hexToBytes(k)).toString();
+    });
+    const keysArr: `0x${string}`[] = _cartItems.size
+      ? Array.from([..._cartItems.keys()])
+      : [];
+    for (const _cartId of keysArr) {
+      const _cart = _cartItems.get(_cartId) as CartState;
+      if (_cart && _cart.status !== IStatus.Failed) {
+        const sig = _cart.signature as `0x${string}`;
+        const retrievedAdd = await relayClient!.recoverSignedAddress(
+          _cartId,
+          sig,
+        );
+        if (addresses.includes(retrievedAdd.toLowerCase())) {
+          console.log("inside inclue", _cartId);
+          setCartId(_cartId);
```

Comment:
does this assume the  clerk has only one cart? or if setCartId gets called multiple times do the carts get saved some where else?

Reply:
Clerk can only have one active cart so setCartId will can get called multiples times but the most recent will be set